### PR TITLE
[Filing home] routing based on user progress in the Filing process

### LINKED ISSUE #534: [Filing Home] Routing based on Submission state
I'm working to ensure that I've accounted for all possible Submission states and how the user should be route on the Frontend accordingly. 

Possible Submission states: https://github.com/cfpb/sbl-filing-api/blob/main/src/sbl_filing_api/entities/models/model_enums.py#L11-L22
 
  ## Upload page - Backend unable to process file; Validations not yet complete so no errors/warnings to show
  SUBMISSION_STARTED
  SUBMISSION_UPLOAD_MALFORMED // format-errors.csv - wrong number columns
  SUBMISSION_UPLOADED
  UPLOAD_FAILED
  VALIDATION_IN_PROGRESS
  VALIDATION_ERROR // Alf picture - file couldn't be processed
  VALIDATION_EXPIRED // Took too long, Need to reupload

  ## Errors page - SBLAR content errors identified
  VALIDATION_WITH_ERRORS

  ## Warnings page - Warnings exist and have not been accepted
  VALIDATION_WITH_WARNINGS

  ## Point of Contact page - No errors, warnings have been accepted if they exist
  VALIDATION_SUCCESSFUL

## Sign and Submit - All previous steps complete
VALIDATION_SUCCESSFUL and Filing has contact info populated

### PR DESCRIPTION
**Pull request by meissadia**
Closes #534 

## Changes
- Filing app routing now accounts for all Submission states
- Updated page content with the latest copy
- Makes additional layout adjustments to Filing home page and InstitutionCards
- [InstitutionCards] Heading H5
- [Global] More accurate spacing - 15.008px (0.938rem) => 15px (0.9375rem)

## How to test
- [See comment below](https://github.com/cfpb/sbl-frontend/pull/535#issuecomment-2116274366).

### CODE REVIEW

**Comment by meissadia** [comment]
Code to test all scenarios.  
 
- Plug the following into the top of `getFilingSteps`. 
- Uncomment each `currentSubmission.state = ` line one at a time
- Verify the Filing homepage links to the correct Filing step page

```
export const getFilingSteps = (
  currentSubmission_: SubmissionResponse,
  currentFiling_: FilingType,
): { filingSteps: StepType[]; nextStepIndex: number } => {
  const currentSubmission = {};
  const currentFiling = { contact_info: false }; // true = POC completed

  // Validations not yet complete
  currentSubmission.state = 'SUBMISSION_STARTED';
  // currentSubmission.state = 'SUBMISSION_UPLOADED';
  // currentSubmission.state = 'SUBMISSION_UPLOAD_MALFORMED';
  // currentSubmission.state = 'VALIDATION_IN_PROGRESS';
  // currentSubmission.state = 'UPLOAD_FAILED';
  // currentSubmission.state = 'VALIDATION_EXPIRED';
  // currentSubmission.state = 'VALIDATION_ERROR';

  // Errors page
  // currentSubmission.state = 'VALIDATION_WITH_ERRORS';

  // Warnings page
  // currentSubmission.state = 'VALIDATION_WITH_WARNINGS';

  // Point of Contact page
  // currentSubmission.state = 'VALIDATION_SUCCESSFUL';

  // Sign and submit page
  // currentFiling.contact_info = true
  // currentSubmission.state = 'SUBMISSION_ACCEPTED';
``

File: src/pages/Filing/FilingApp/FilingOverviewPage.tsx [hunk 1]
```diff
@@ -54,33 +54,30 @@ export default function FilingOverview(): ReactElement {
   const defaultFilingPeriod = filingPeriods?.[0]?.code ?? currentYear;
 
   return (
-    <div className='u-mt45 mx-auto max-w-[41.875rem]'>
+    <div className='u-mt45 mx-auto max-w-[48.25rem]'>
```
[the remark this answers is not in the document]
Comment (meissadia) [reply src/pages/Filing/FilingApp/FilingOverviewPage.tsx]: Great catch, updated.

File: src/pages/Filing/FilingApp/InstitutionCard.tsx [hunk 2]
```diff
@@ -26,21 +27,14 @@ function SecondaryButton({
...
   return (
-    <span className='ml-[0.938rem] inline-block'>
-      <Button
-        asLink
-        label={secondaryButtonLabel}
-        onClick={onSecondaryClick}
-        className='ml-2'
-      />
-    </span>
+    <p className='ml-[0.938rem] inline-block font-medium'>
```
[the remark this answers is not in the document]
Comment (meissadia) [reply src/pages/Filing/FilingApp/InstitutionCard.tsx]: Good call, globally adjusted in this PR.

**Comment by meissadia** [comment]
> It's been working well while I've been testing. 👍
> 
> We will eventually need something like I did for the shared lending platform: prevent users from skipping ahead via the url unless `toggleRouting` is set to true.
> 
> Something for later though! 👍

https://github.com/cfpb/sbl-frontend/issues/582

**Review by billhimmelsbach — APPROVED** [review]
Awesome! Thanks for ticketing the routing.

**Pull request merged**